power_grid_model_c/power_grid_model_c/src/model.cpp

```diff
@@ -76,36 +76,36 @@ void PGM_calculate(PGM_Handle* handle, PGM_PowerGridModel* model, PGM_Options co
 
     // call calculation
     try {
-        auto const calculation_method = static_cast<CalculationMethod>(opt->calculation_method);
+        auto const options =
+            MainModel::Options{.calculation_method = static_cast<CalculationMethod>(opt->calculation_method),
+                               .err_tol = opt->err_tol,
+                               .max_iter = opt->max_iter,
+                               .threading = opt->threading,
+                               .short_circuit_voltage_scaling =
+                                   static_cast<ShortCircuitVoltageScaling>(opt->short_circuit_voltage_scaling)};
+
         switch (opt->calculation_type) {
         case PGM_power_flow:
             if (opt->symmetric != 0) {
```

Comment:
moving `opt->calculation_type` and `opt->symmetric` also into the `MainModel::Options` can be done at a later stage